src/compiler/checker.ts

```diff
@@ -7013,12 +7014,22 @@ namespace ts {
 
         // Return the contextual type for a given expression node. During overload resolution, a contextual type may temporarily
         // be "pushed" onto a node using the contextualType property.
-        function getContextualType(node: Expression): Type {
-            let type = getContextualTypeWorker(node);
+        function getApparentTypeOfContextualType(node: Expression): Type {
+            let type = getContextualType(node);
             return type && getApparentType(type);
         }
 
-        function getContextualTypeWorker(node: Expression): Type {
+        /**
+         * Woah! Do you really want to use this function?

```

Comment:
Ah, that makes sense, though it feels strange to get the actual contextual type when its name is worker.

If you think it'd be better to rename it back to worker, I can do that.

I also suspect that we could shave off some time by using this function (instead of `getApparentTypeOfContextualType`) for array literals as well. I'll probably do it as part of another change.